src/main/java/com/spotify/annoy/ANNIndex.java

```diff
@@ -8,53 +8,69 @@
 import java.util.*;
 
 /**
- * Read-only Approximate Nearest Neighbor Index which queries databases created by annoy.
+ * Read-only Approximate Nearest Neighbor Index which queries
+ * databases created by annoy.
  */
-public class ANNIndex implements AnnoyIndex {
```

Comment:
I don't like having an interface for a single class implementing it, it suggests the idea that there might be different implementations of an index when in fact there aren't.
I'd rather change the internal code to avoid having the need for it, though I won't object if you guys do in fact really want to keep it in there.

It would be different if we had two classes, say an `AngularIndex` and an `EuclideanIndex` class but I'd rather keep things simple.